### ISSUE TITLE
RustGlue hardcodes the name `mainForHost`

### ISSUE THREAD

**Issue by rtfeldman** [roc-lang/roc#5693]
This is currently done as a workaround for an issue; the fix should involve fixing the problem noted in https://github.com/roc-lang/roc/pull/5688/commits/c8b518df88dba10cc45bd00afe8ecd19425bc8e1

**Comment by skyqrose** [comment]
The name in the linked commit is still hardcoded. Now it's at https://github.com/roc-lang/roc/blob/69dd8d77f3127cb21eaa34f8c7533930e30b923a/crates/glue/src/types.rs#L1272 , without the comment.

There was at least one other place where "mainForHost" was hardcoded that was recently fixed in https://github.com/roc-lang/roc/pull/7147 (on one of Brendan's streams: https://www.youtube.com/watch?v=bANcw60Y-II ). I'm not sure whether that means the underlying issue is now fixed, or if it was completely unrelated.

**Comment by lukewilliamboswell** [comment]
I think the issue still remains, though this is a low priority. The glue generated by `RustGlue.roc` is not currently usable for the platform entry points, so for all of the related platform this part is written by hand. This is also much easier to do by hand with the removal of `Task` wrapping in the purity inference change. 

So once purity-inference lands, I think resolving this and having RustGlue.roc generate the correct types for entry points will be much easier.
